### PR TITLE
Pass over federation API section

### PR DESCRIPTION
**Pull request by kkohbrok**
A general pass over the API section to bring it up to date, reduce the level of detail and try to give a good high-level overview over what's going on. I'd greatly appreciate feedback on Section 4.6, where I put my best guess of how I would imagine the flows to work. Also, some feedback on how to bring Section 4.4 up to date would be great.

## Checklist:

Please tick the following before making your PR:

* [x] I ran `make` on this branch to build the docs and there are **no WARNINGs**.
* [x] After running `make`, I looked at the output by opening build/html/index.html in a browser and I'm satisfied with the rendering.

### CODE REVIEW

File: src/understand/federation/api.rst [hunk 1]
```diff
@@ -166,97 +155,119 @@ The federator connects to brig and makes an HTTP request which looks like this:
...
+
+1. From the search results of a :ref:`user discovery<user-discovery>` process,
+   `A` chooses to create a conversation with `B`.
+2. `A`'s client issues a ``/users/backend-b.com/B/prekeys`` query to `A`'s
+   backend.
+3. `A`'s backend queries the ``claim-prekey-bundle`` endpoint of `B`'s backend
+   using `B`'s user id.
+4. `B`'s backend replies with a prekey bundle for each of `B`'s clients.
+5. `A`'s backend forwards that information to `A`'s client.
+6. `A`'s client queries the ``/conversations/one2one`` endpoint of its backend
```
Comment (jschaul) [inline src/understand/federation/api.rst:270]: At this stage, only group conversations are supported, the one2one flow is a bit more complicated as it involves connections which need to be accepted (this isn't implemented yet). So the flow to document here would be the one for a group conversation, using the `POST /conversations` endpoint (containing B's qualified user ID).
```suggestion
6. `A`'s client queries the ``/conversations`` endpoint of its backend
```

**Review by jschaul — APPROVED** [review]
Looks good! I pushed one commit about a slight change in the example.

File: src/understand/federation/api.rst [hunk 2]
```diff
@@ -154,109 +143,134 @@ this (shown as JSON for convenience):
 
    {
      "component": "Brig",
-     "method": "GET",
-     "path": "/users/by-handle",
-     "query": [ { "key": "handle", "value": "janedoe" } ],
-     "body": null
+     "path": "federation/get-user-by-handle",
+     "body": "\"janedoe\""
```
Comment (pcapriotti) [inline src/understand/federation/api.rst:147]:
```suggestion
     "body": "\"janedoe\"",
```

**Review by pcapriotti — APPROVED** [review]
All pretty accurate. Small typo fix below.

**Pull request merged**